### ISSUE TITLE
Recurring contribution triggers DB Error: constraint violation

### ISSUE THREAD

**Issue by h-c-c** [drastik/com.drastikbydesign.stripe#117]
My test setup is using:
Drupal 7.38
CiviCRM 4.6.8
Stripe extension 4.6-dev cloned 11/19
API version 2015-10-16 

My extension was updated, so the database tables would have been created by an earlier version.  I'm not sure how that factors in, if at all.

```

Nov 21 05:30:49  [info] $Fatal Error Details = Array
(
    [callback] => Array
        (
            [0] => CRM_Core_Error
            [1] => handle
        )

    [code] => -3
    [message] => DB Error: constraint violation
    [mode] => 16
    [debug_info] => INSERT INTO civicrm_contribution (
          contact_id, financial_type_id, payment_instrument_id, receive_date,
          total_amount, fee_amount, net_amount, trxn_id, invoice_id, currency,
          contribution_recur_id, is_test, contribution_status_id, campaign_id
          ) VALUES (
          2, 2, 1, '2015-11-21 05:30:39',
          '1', '0', '1', 'ch_179123456789876543210', 'in_17909876564321234567', 'USD',
          6, 0, '1', 0) [nativecode=1452 ** Cannot add or update a child row: a foreign key constraint fails (`civicrm`.`civicrm_contribution`, CONSTRAINT `FK_civicrm_contribution_campaign_id` FOREIGN KEY (`campaign_id`) REFERENCES `civicrm_campaign` (`id`) ON DELETE SET NULL)]
    [type] => DB_Error
    [user_info] => INSERT INTO civicrm_contribution (
          contact_id, financial_type_id, payment_instrument_id, receive_date,
          total_amount, fee_amount, net_amount, trxn_id, invoice_id, currency,
          contribution_recur_id, is_test, contribution_status_id, campaign_id
          ) VALUES (
          2, 2, 1, '2015-11-21 05:30:39',
          '1', '0', '1', 'ch_179123456789876543210', 'in_17909876564321234567', 'USD',
          6, 0, '1', 0) [nativecode=1452 ** Cannot add or update a child row: a foreign key constraint fails (`civicrm`.`civicrm_contribution`, CONSTRAINT `FK_civicrm_contribution_campaign_id` FOREIGN KEY (`campaign_id`) REFERENCES `civicrm_campaign` (`id`) ON DELETE SET NULL)]
    [to_string] => [db_error: message="DB Error: constraint violation" code=-3 mode=callback callback=CRM_Core_Error::handle prefix="" info="INSERT INTO civicrm_contribution (
          contact_id, financial_type_id, payment_instrument_id, receive_date,
          total_amount, fee_amount, net_amount, trxn_id, invoice_id, currency,
          contribution_recur_id, is_test, contribution_status_id, campaign_id
          ) VALUES (
          2, 2, 1, '2015-11-21 05:30:39',
          '1', '0', '1', 'ch_179123456789876543210', 'in_17909876564321234567', 'USD',
          6, 0, '1', 0) [nativecode=1452 ** Cannot add or update a child row: a foreign key constraint fails (`civicrm`.`civicrm_contribution`, CONSTRAINT `FK_civicrm_contribution_campaign_id` FOREIGN KEY (`campaign_id`) REFERENCES `civicrm_campaign` (`id`) ON DELETE SET NULL)]"]
)


Nov 21 05:30:49  [info] $backTrace = #0 /var/www/drupal-7.38/sites/all/modules/civicrm/CRM/Core/Error.php(235): CRM_Core_Error::backtrace("backTrace", TRUE)
#1 [internal function](): CRM_Core_Error::handle(Object(DB_Error))
#2 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/PEAR.php(931): call_user_func((Array:2), Object(DB_Error))
#3 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/DB.php(976): PEAR_Error->PEAR_Error("DB Error: constraint violation", -3, 16, (Array:2), "INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...")
#4 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/PEAR.php(564): DB_Error->DB_Error(-3, 16, (Array:2), "INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...")
#5 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/DB/common.php(1905): PEAR->raiseError(NULL, -3, NULL, NULL, "INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...", "DB_Error", TRUE)
#6 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/DB/mysql.php(899): DB_common->raiseError(-3, NULL, NULL, NULL, "1452 ** Cannot add or update a child row: a foreign key constraint fails (`ci...")
#7 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/DB/mysql.php(328): DB_mysql->mysqlRaiseError()
#8 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/DB/common.php(1216): DB_mysql->simpleQuery("INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...")
#9 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/DB/DataObject.php(2442): DB_common->query("INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...")
#10 /var/www/drupal-7.38/sites/all/modules/civicrm/packages/DB/DataObject.php(1634): DB_DataObject->_query("INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...")
#11 /var/www/drupal-7.38/sites/all/modules/civicrm/CRM/Core/DAO.php(325): DB_DataObject->query("INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...")
#12 /var/www/drupal-7.38/sites/all/modules/civicrm/CRM/Core/DAO.php(1179): CRM_Core_DAO->query("INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...", TRUE)
#13 /var/www/drupal-7.38/sites/default/files/civicrm/extensions/com.drastikbydesign.stripe/CRM/Stripe/Page/Webhook.php(141): CRM_Core_DAO::executeQuery("INSERT INTO civicrm_contribution (\n          contact_id, financial_type_id, ...", (Array:13))
#14 /var/www/drupal-7.38/sites/all/modules/civicrm/CRM/Core/Invoke.php(312): CRM_Stripe_Page_Webhook->run((Array:3), NULL)
#15 /var/www/drupal-7.38/sites/all/modules/civicrm/CRM/Core/Invoke.php(86): CRM_Core_Invoke::runItem((Array:13))
#16 /var/www/drupal-7.38/sites/all/modules/civicrm/CRM/Core/Invoke.php(54): CRM_Core_Invoke::_invoke((Array:3))
#17 /var/www/drupal-7.38/sites/all/modules/civicrm/drupal/civicrm.module(489): CRM_Core_Invoke::invoke((Array:3))
#18 [internal function](): civicrm_invoke("stripe", "webhook")
#19 /var/www/drupal-7.38/includes/menu.inc(519): call_user_func_array("civicrm_invoke", (Array:2))
#20 /var/www/drupal-7.38/index.php(21): menu_execute_active_handler()
#21 {main}
```

**Comment by h-c-c** [comment]
I guess this has been fixed by @jmcclelland and there's a comment about a pull request.  

![screen shot 2015-11-21 at 2 28 30 pm](https://cloud.githubusercontent.com/assets/5252344/11320081/776badd2-905c-11e5-989a-8cd374d0dc31.png)

Was this pull request actually made?  I don't see that the fix went in, or any outstanding pull request for it.  

https://github.com/progressivetech/com.drastikbydesign.stripe/commit/e28fda6820b1af1657043cb25b5061de05973b5f

Thanks much!

**Comment by jmcclelland** [comment]
Thanks for noticing this - I didn't realize that my last fix for #92 was never merged. I've created a new pull request: https://github.com/drastik/com.drastikbydesign.stripe/pull/118 that was made against the 4.6-dev branch.

**Comment by h-c-c** [comment]
No prob!  I'm not sure about 92.....I'll comment over there.
